### PR TITLE
Update max height previously forged - Closes #3696

### LINKED ISSUE #3696: Update the block processing to persist maxHeightPreviouslyForged
### Expected behavior

Update the block processing to persist maxHeightPreviouslyForged, this value should be updated at the end of the steps of forging a block. 

This property needs to be updated and saved before broadcasting a block.

> If a delegate forges a block B to be added to the current chain, then B.heightPrevious must be the largest height at which the delegate forging the block previously forged a block on any chain or 0 if the delegate did not forge a block before. To be able to provide this information correctly, even in the case of node crashes, it is important that the largest height of a block previously forged is persistently stored before any forged block is broadcast to the network.

### Which version(s) does this affect? (Environment, OS, etc...)

3.0.0

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
- Max height previously forged was not persisted when creating a block
- Wrong maxPreviouslyForgedHeight was used while creating a block

### How did I solve it?
- Add a function to save previously forged height per delegate
- Use the saved value for BFT header

### How to manually test it?
- Start the node and observe chain_meta table that maxHeightPreviouslyForged is saved
- Check that newly created block has the correct previously forged height per delegate

### Review checklist

- [ ] The PR resolves #3696 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/block_processor_v2.js [hunk 1]
```diff
@@ -224,6 +224,13 @@ class BlockProcessorV2 extends BaseBlockProcessor {
 				return this.bftModule.computeBFTHeaderProperties(delegatePublicKey);
 			},
 			(data, bftHeader) => this._create({ ...data, ...bftHeader }),
+			async (data, block) => {
```
Comment (michielmulders) [inline framework/src/modules/chain/block_processor_v2.js]: Can you add a comment above the create.pipe briefly explaining the three steps that are happening in the create pipeline?

File: framework/src/modules/chain/bft/bft.js [hunk 2]
```diff
@@ -151,11 +152,44 @@ class BFT extends EventEmitter {
...
+	 * Saving a height which delegate last forged. this needs to be saved before broadcasting
+	 * so it needs to be outside of the DB transaction
+	 * @param delegatePublicKey
+	 * @param height
+	 */
+	async saveMaxHeightPreviouslyForged(delegatePublicKey, height) {
+		const previouslyForged = await this._getPreviouslyForgedMap();
+		const updatedPreviouslyForged = {
+			...previouslyForged,
+			[delegatePublicKey]: height,
```
Comment (lsilvs) [inline framework/src/modules/chain/bft/bft.js]: Don't we need to check if `previouslyForged[delegatePublicKey].height` is less than `height` before updating?

Comment (shuse2) [reply framework/src/modules/chain/bft/bft.js]: I think it should always update to the last forging block regardless of height

Comment (shuse2) [reply framework/src/modules/chain/bft/bft.js]: @lsilvs You are right. It's maxPreviouslyForgedHeight so it should check (confirmed with research team)

**Pull request merged**